.template/lib/main.dart

```diff
@@ -2,34 +2,53 @@ import 'package:flutter/material.dart';
 import 'package:flutter_config/flutter_config.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 import 'package:flutter_templates/gen/assets.gen.dart';
+import 'package:go_router/go_router.dart';
 import 'package:package_info_plus/package_info_plus.dart';
 
 void main() async {
   WidgetsFlutterBinding.ensureInitialized();
   await FlutterConfig.loadEnvVariables();
-  runApp(const MyApp());
+  runApp(MyApp());
 }
 
 class MyApp extends StatelessWidget {
-  // TODO: implement Routes then remove `home: MyHomePage()` and use initialRoute instead.
-  final String initialRoute;
+  MyApp({Key? key}) : super(key: key);
 
-  const MyApp({
-    Key? key,
-    this.initialRoute = '/',
-  }) : super(key: key);
+  final GoRouter _router = GoRouter(
+    routes: <GoRoute>[
+      GoRoute(
+        path: '/',
+        builder: (BuildContext context, GoRouterState state) =>
+            const MyHomePage(),
+        routes: [
+          GoRoute(
+            path: 'page2',
+            builder: (BuildContext context, GoRouterState state) =>
+                const Page2Screen(),
+          ),
+          GoRoute(
+            path: 'page3',
```

Comment:
Should we define some route path constants?

Reply:
Fixed in 5a789646d0597aa3c6719c36c9fe667e256ec780